src/Switch.jsx

```diff
@@ -86,20 +84,21 @@ class Switch extends Component {
 
   render() {
     const { className, prefixCls, disabled, loadingIcon,
-      checkedChildren, tabIndex, unCheckedChildren, ...restProps } = this.props;
+      checkedChildren, unCheckedChildren, ...restProps } = this.props;
     const checked = this.state.checked;
-    const switchTabIndex = disabled ? -1 : (tabIndex || 0);
     const switchClassName = classNames({
       [className]: !!className,
       [prefixCls]: true,
       [`${prefixCls}-checked`]: checked,
       [`${prefixCls}-disabled`]: disabled,
     });
     return (
-      <span
+      <button
         {...restProps}
```

Comment:
I think you should add  type=button because by default, if it's used in a form, pressing enter will trigger a submit.

Reply:
good call, will do